### DISCUSSION TITLE
The new fastapi-cli is in circler dependency with the fastapi - this breaks bazel builds

### DISCUSSION THREAD

**Discussion by ns-ggeorgiev** [fastapi/fastapi#11529] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
fastapi==0.111.0 \
    --hash=sha256:97ecbf994be0bcbdadedf88c3150252bed7b2087075ac99735403b1b76cc8fc0 \
    --hash=sha256:b9db9dd147c91cb8b769f7183535773d8741dd46f9dc6676cd82eab510228cd7
    # via
    #   fastapi-cli
    #   pact-python
fastapi-cli==0.0.2 \
    --hash=sha256:589565ba758432632eadcf7b950e0ec76bb283b549784d9df17f261a8a9de476 \
    --hash=sha256:d7a8ec89fd52ad16c52de9fe7299e4b22c7a32e1bf28aa886e5517e4927423dd
    # via fastapi
```

### Description

Bazel is very opinionated build system and does not allow for circler dependency. This unwise decision to create to libraries depending on each other will likely be breaking a lot of build when they attempt to update to the new version.

### Operating System

Linux

### Operating System Details

it does not matter

### FastAPI Version

111

### Pydantic Version

does not matter

### Python Version

does not matter

**Comment by brunobely** [comment]
Doesn't even need fastapi-cli as a dep. Just fastapi breaks my builds.

**Reply by blurhead** [reply]
`d71be59` looks like this commit cause this issue

**Reply by Kyle-sandeman-mrdfood** [reply]
I can confirm that my existing `requirements.txt` with `fastapi>=0.109.0` has started to try multiple versions of uvicorn and other libs and ends up taking super long. Adding `<0.110.0` fixes it. This is via standard `pip install -r requirements.txt` with most dependencies unpinned

**Reply by Kyle-sandeman-mrdfood** [reply]
Actually this seems like a really weird edge-case.
It only happens when I install `click<8` alongside, but that package has no dependencies (??)
To reproduce
```shell
python -m venv venv; source venv/bin/activate
python -V; pip -V  # Python 3.12.1 pip 23.2.1
pip install 'fastapi[all]>=0.109.0' 'click<8'
```
And eventually:
```
INFO: pip is looking at multiple versions of typer to determine which version is compatible with other requirements. This could take a while.
Collecting uvicorn[standard]>=0.12.0 (from fastapi[all]>=0.109.0)
  Obtaining dependency information for uvicorn[standard]>=0.12.0 from https://files.pythonhosted.org/packages/73/f5/cbb16fcbe277c1e0b8b3ddd188f2df0e0947f545c49119b589643632d156/uvicorn-0.29.0-py3-none-any.whl.metadata
  Downloading uvicorn-0.29.0-py3-none-any.whl.metadata (6.3 kB)
  ...
```

Edit: turns out `uvicorn` depends on `click`...

**Comment by tiangolo** [comment, marked as answer]
Thanks for the feedback! I checked that Python indeed supported circular dependencies before setting it up, but seeing that Bazel (and maybe other systems) would handle dependencies in a more strict way, I removed the circular dependency requirement. :nerd_face: 

This is released in FastAPI CLI 0.0.4 :tada:
